### ISSUE TITLE
Templates for mount units

### ISSUE THREAD

**Issue by vieira** [systemd/systemd#3504]
### Submission type
- [ ] Bug report
- [X] Request for enhancement (RFE)
### systemd version the issue has been seen with

230
### Used distribution

Arch
### In case of bug report: Expected behaviour you didn't see

I expected templates to work with mount units, so I could make `systemd-nspawn@container` only launch after `var-lib-machines-@container.mount` which is an overlayfs. This is needed because automount does not work with overlay (see #3076). There also seems to be some interest from the coreos people (https://github.com/coreos/fleet/issues/1390).
### In case of bug report: Unexpected behaviour you saw

Assertion 'unit' failed at src/shared/bus-unit-util.c:1245, function unit_show_processes(). Aborting
### In case of bug report: Steps to reproduce the problem

```
$ mkdir -p /var/lib/{machines/test,images/test,deltas/test}
$ cat /etc/systemd/system/var-lib-machines-\@.mount 
[Mount]
What=overlay
Where=/var/lib/machines/%i
Type=overlay
Options=upperdir=/var/lib/machines/%i,lowerdir=/var/lib/images/test,workdir=/var/lib/deltas/%i
```

```
$ systemctl status var-lib-machines-@test.mount
Assertion 'unit' failed at src/shared/bus-unit-util.c:1245, function unit_show_p
● n/a
   Loaded: n/a
   Active: n/a
Aborted (core dumped)
```

**Comment by poettering** [comment]
Uh, there are two things here I figure... The assertion, plus an RFE.

**Comment by poettering** [comment]
The assertion fix is waiting in #3527.

**Comment by poettering** [comment]
With #8271 we have an equivalent concept implemented now. Closing this one.
